src/FuseGPUThreadLoops.cpp

```diff
@@ -302,15 +385,15 @@ class ExtractSharedAllocations : public IRMutator {
             return;
         }
 
-        shared.insert(op->name);
+        shared.emplace(op->name, IntInterval(barrier_stage, barrier_stage));
         IRMutator::visit(op);
-        shared.erase(op->name);
         op = stmt.as<Allocate>();
         internal_assert(op);
 
         SharedAllocation alloc;
         alloc.name = op->name;
         alloc.type = op->type;
+        alloc.liveness = shared[op->name];
```

Comment:
It looks like shared[op->name].max gets updated below when you hit a load or store, but I don't see where alloc.liveness gets updated.